edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/KubernetesModule.cs

```diff
@@ -340,6 +343,10 @@ protected override void Load(ContainerBuilder builder)
                     })
                 .As<Task<IEnvironmentProvider>>()
                 .SingleInstance();
+
+            // SystemResourcesMetrics
+            builder.Register(c => new Edgelet.Docker.SystemResourcesMetrics(c.Resolve<IMetricsProvider>(), c.Resolve<IModuleManager>().GetSystemResourcesAsync, this.apiVersion, this.performanceMetricsUpdateFrequency))
```

Comment:
It might be better to make a ISystemResourcesMetrics interface and a NullSystemResourcesMetrics, since SystemResourcesMetrics makes http calls to docker. If the enable metrics flag is set to true it will try to start this in docker, and I'm not sure what would happen. 

Alternatively, you could replace `c.Resolve<IModuleManager>().GetSystemResourcesAsync` with `() => Task.FromResult(new Edgelet.Models.SystemResources(0, 0, 0, 0, 0, new Edgelet.Models.Disk[0], string.Empty))`. This will prevent the call to edgelet

Reply:
Made an ISystemResourcesMetrics and an empty implementation for k8s.